safe-global/safe-wallet-web
# fix: session management when switching chains

## What it solves

Resolves unsupported sessions remaining connected

## How this PR fixes it

1. When switching chain, DApps that don't support the chain being opened are disconnected.
2. If navigating out of a "Safe route", e.g. the welcome page, all sessions are disconnected.

## How to test it

1. Open a mainnet Safe and connect to AAVE and mint.fun. Observe mint.fun disconnecting when switching to a Polygon Safe.
2. Navigate to the welcome page and observe AAVE disconnecting, as well as the WC button disabled.

## Screenshots

![connection](https://github.com/safe-global/safe-wallet-web/assets/20442784/f9006aa2-d1fe-4fe7-a7e0-70799ca92a20)

## Checklist
